### DISCUSSION TITLE
Internal SDK or Framework of PyPDF2

### DISCUSSION THREAD

**Discussion by VedhaGanesh** [py-pdf/pypdf#1422] [Q&A] [answered]
Hello, I just wanted to know what technology or SDK or Framework does PyPDF2 internally use?
I want to contribute to open source PyPDF2 so I need basic help with it.

**Comment by MartinThoma** [comment, marked as answer]
PyPDF2 uses python. Nothing more.*

* There are some exceptions : cryptodome for cryptography and pillow for handing images. And testing is a completely different story.
